docs/catalog-index-configuration.md

```diff
@@ -0,0 +1,103 @@
+# Catalog Index Configuration
+
+The catalog index is an OCI artifact that contains `dynamic-plugins.default.yaml`, which defines the default set of dynamic plugins to be installed. The Helm chart automatically configures the `install-dynamic-plugins` init container to pull and extract this catalog index.
+
+By default, the chart sets the `CATALOG_INDEX_IMAGE` environment variable in the `install-dynamic-plugins` init container:
+
+```yaml
+env:
+  - name: CATALOG_INDEX_IMAGE
+    value: "quay.io/rhdh/plugin-catalog-index:1.9"
+```
+
+The `install-dynamic-plugins.py` script:
+1. Pulls the catalog index OCI image using `skopeo`
+2. Extracts the image layers to a temporary directory (`.catalog-index-temp`)
+3. Locates `dynamic-plugins.default.yaml` within the extracted content
+4. Replaces the `dynamic-plugins.default.yaml` reference in your `includes` list with the extracted catalog index version
```

Comment:
Seems redundant with the docs in the rhdh repo. As discussed, we can probably link to https://github.com/redhat-developer/rhdh/blob/main/docs/dynamic-plugins/installing-plugins.md#using-a-catalog-index-image-for-default-plugin-configurations for further details.
We should only put the information specific to the Helm Chart here. And later update https://github.com/redhat-developer/rhdh/blob/main/docs/dynamic-plugins/installing-plugins.md#using-a-catalog-index-image-for-default-plugin-configurations with links to the respective Helm chart and Operator repos with more information on how to configure the catalog index image.